tests/lir_tests/convert_lir.rs

```diff
@@ -0,0 +1,481 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ffi::OsStr;
+
+use solang::{
+    codegen::codegen,
+    file_resolver::FileResolver,
+    lir::{converter::Converter, printer::Printer},
+    parse_and_resolve,
+    sema::ast::Namespace,
+    Target,
+};
+
+use crate::stringfy_lir;
+
+fn new_file_resolver(src: &str) -> FileResolver {
+    let mut cache = FileResolver::default();
+    cache.set_file_contents("test.sol", src.to_string());
+    cache
+}
+
+fn assert_lir_str_eq(src: &str, cfg_no: usize, expected: &str) {
+    let mut resolver = new_file_resolver(src);
+    let mut ns: Namespace =
+        parse_and_resolve(OsStr::new("test.sol"), &mut resolver, Target::Solana);
+    // check errors
+    // if !ns.diagnostics.is_empty() {
```

Comment:
There's a bunch of commented out code here. Commented out code should rarely be kept and if so it should come with a comment explaining why it's commented out but not deleted from the code base.

Reply:
I eventually removed the commented code. This is helpful for debugging unit test failures.
```
if !ns.diagnostics.is_empty() {
    ns.print_diagnostics_in_plain(&resolver, false);
}
```